instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -121,18 +121,11 @@ def request_span_attributes(env:)
               'http.method' => env['REQUEST_METHOD'],
               'http.host' => env['HTTP_HOST'] || 'unknown',
               'http.scheme' => env['rack.url_scheme'],
-              'http.target' => fullpath(env)
+              'http.target' => env['QUERY_STRING'].empty? ? env['PATH_INFO'] : "#{env['PATH_INFO']}?#{env['QUERY_STRING']}"
             }
-            attributes['http.user_agent'] = env['HTTP_USER_AGENT'] if env['HTTP_USER_AGENT']
-            attributes.merge(allowed_request_headers(env))
-          end
 
-          # e.g., "/webshop/articles/4?s=1":
-          def fullpath(env)
-            query_string = env['QUERY_STRING']
-            path = env['SCRIPT_NAME'] + env['PATH_INFO']
-
-            query_string.empty? ? path : "#{path}?#{query_string}"
+            attributes['http.user_agent'] = env['HTTP_USER_AGENT'] if env['HTTP_USER_AGENT']
```

Comment:
Have we decided to start using [Semantic Conventions Constants](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/semantic_conventions/lib/opentelemetry/semantic_conventions/trace.rb#L162) everywhere? If so should we update this as part of this change?

Reply:
>Have we decided to start using Semantic Conventions Constants everywhere? 

I'm not sure actually, I'll bring it up at the next sig to see what we want to do there.

>If so should we update this as part of this change?

I think this change would happen as a "big sweeping change" across the gems.  So I'd prefer to keep it out of this PR.

Reply:
> > Have we decided to start using Semantic Conventions Constants everywhere?
> 
> I'm not sure actually, I'll bring it up at the next sig to see what we want to do there.

Yes, we should use the Semantic Conventions Constants everywhere.

> 
> > If so should we update this as part of this change?
> 
> I think this change would happen as a "big sweeping change" across the gems. So I'd prefer to keep it out of this PR.

Agreed